.babelrc

```diff
@@ -16,5 +16,24 @@
         "loose": true
       }
     ]
-  ]
+  ],
+
+  "env": {
+    "development": {
+      "plugins": ["external-helpers"]
+    },
+
+    "production": {
+      "plugins": [
+        "external-helpers",
+        "transform-react-inline-elements",
+        "transform-react-constant-elements",
```

Comment:
>  if the minifier support inlining of invariants:

Well it seems uglifyjs doesn't do this in the current setup, I remember that Google Closure compiler did this only in advanced mode.

But in our case I'd say we don't want to inline code related to jsx to keep as less memory allocations as possible. I can think of something like this:
```
// Code :
export default class Component extends React.Component {
  render() {
    const markup = <hr className="hr" />;
    return markup;
  }
}

// After Babel
var markup = React.createElement("hr", { className: "hr" });
var Component = function (_React$Component) {
  ........
  Component.prototype.render = function render() {
    return markup;
  };
  ........
}(React.Component);
``` 

So we share this immutable `markup` between instances. 

But №2 :) Just looked more closely and I don't see that `transform-react-constant-elements` changes anything in compiled version when `transform-react-inline-elements` is used... Didn't spend time to understand, I'd rather keep them both just in case, but can leave only `transform-react-inline-elements` if you want @gmarty ?